tsdb/head.go

```diff
@@ -373,32 +380,40 @@ func NewHead(r prometheus.Registerer, l log.Logger, wal *wal.WAL, opts *HeadOpti
 		opts.SeriesCallback = &noopSeriesLifecycleCallback{}
 	}
 
-	es, err := NewCircularExemplarStorage(opts.NumExemplars, r)
-	if err != nil {
-		return nil, err
+	var em *ExemplarMetrics
+	// Always start with a noopExemplarStorage now since ApplyConfig is run right after TSDB starts up.
+	var es ExemplarStorage = noopExemplarStorage{}
+	if opts.EnableExemplarStorage && opts.MaxExemplars > -1 {
```

Comment:
```suggestion
	if opts.EnableExemplarStorage {
```

Because we are taking care of <=0 in `NewCircularExemplarStorage` and might be best to leave it to the exemplar store implementation to decide the optimisations for such things.